### PR TITLE
feat(phone): `Phone` supports national format

### PR DESCRIPTION
**Pull request by jbenning-ic**
## Summary

Adds a new `format` option to `maskitoPhoneOptionsGenerator` that supports `'NATIONAL'` and `'INTERNATIONAL'` formats (default: `'INTERNATIONAL'`).

Closes: https://github.com/taiga-family/maskito/issues/1940

When `format: 'NATIONAL'` is used with strict mode:
- Phone numbers are formatted without country code prefix (e.g., `(213) 373-4253` for US instead of `+1 213 373-4253`)
- Uses libphonenumber-js `formatNational()` for proper national formatting

image

## Usage

```typescript
import {maskitoPhoneOptionsGenerator} from '@maskito/phone';
import metadata from 'libphonenumber-js/min/metadata';

const mask = maskitoPhoneOptionsGenerator({
  countryIsoCode: 'US',
  metadata,
  format: 'NATIONAL', // Format as (212) 343-3355 instead of +1 212 343-3355
});
```

### CODE REVIEW

**Comment by DevWedeloper** [comment]
I implemented something like this for my own use-case. It would be great to have this out of the box, hopefully this PR get merged

**Comment by jbenning-ic** [comment]
Hi @nsbarsukov, hope you are doing well! 

I know this PR is rather large and the holidays are almost here, but I was just curious to know if you have a rough estimate as to when we can expect a review on this PR? 

We just need to know whether we can expect this feature in early 2026, or if we will need to develop our own util in the meantime. 

Thank you again for all of the great work and have a happy holidays 🙏

**Comment by nsbarsukov** [comment]
@jbenning-ic Hello!
I’m afraid I won't have time to review it this week. I’ll try my best, but I can't make any promises. 
After that, I am on holiday for two weeks completely offline. I plan to review it during the week of January 12–16.

**Comment by jbenning-ic** [comment]
@nsbarsukov Thank you very much for the info! 

I hope you enjoy your holiday, and have a great start to 2026 🚀

File: projects/demo/src/pages/phone/phone-doc.component.ts [hunk 1]
```diff
@@ -131,6 +151,14 @@ export default class PhoneDocComponent implements GeneratorOptions {
...
 
+        /**
+         * For national format, don't add focus/blur plugins since there's no prefix.
+         * For international strict mode, add focus/blur plugins for prefix handling.
+         */
+        if (this.format === 'NATIONAL') {
+            return options;
+        }
+
         return this.strict
```
Comment (nsbarsukov) [inline projects/demo/src/pages/phone/phone-doc.component.ts]:
```suggestion
        return this.strict && this.format === 'INTERNATIONAL'
```

File: projects/phone/src/lib/masks/phone/phone-mask.ts [hunk 2]
```diff
@@ -4,19 +4,39 @@ import type {CountryCode, MetadataJson} from 'libphonenumber-js/core';
 import {maskitoPhoneNonStrictOptionsGenerator} from './phone-mask-non-strict';
 import {maskitoPhoneStrictOptionsGenerator} from './phone-mask-strict';
 
+/**
+ * Phone number format type.
+ * - 'INTERNATIONAL': Includes country code prefix (e.g., +1 212 343-3355)
+ * - 'NATIONAL': Country-specific format without country code (e.g., (212) 343-3355)
+ */
+export type PhoneNumberFormat = 'INTERNATIONAL' | 'NATIONAL';
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/phone-mask.ts]: I am not sure that we really need new type.

Let's implement the same approach as used for `@maskito/kit` generators (for code-style consistency across libraries):

```ts
export function maskitoNumberOptionsGenerator(
    params: MaskitoNumberParams
): Required<MaskitoOptions>;

export function maskitoDateOptionsGenerator(
    params: MaskitoDateParams
): Required<MaskitoOptions>;

// [...]

export interface MaskitoPhoneParams {
    countryIsoCode?: CountryCode;
    metadata: MetadataJson;
    strict?: boolean;
    separator?: string;
    /**
     * Phone number format.
     * - 'INTERNATIONAL' (default): Includes country code prefix (e.g., +1 212 343-3355)
     * - 'NATIONAL': Country-specific format without country code (e.g., (212) 343-3355).
     *   Only works with strict mode (requires countryIsoCode).
     */
    format?: 'NATIONAL' | 'INTERNATIONAL';
}

export function maskitoPhoneOptionsGenerator(
    params: MaskitoPhoneParams
): Required<MaskitoOptions>;
```

Comment (nsbarsukov) [reply projects/phone/src/lib/masks/phone/phone-mask.ts]: Or even
```ts
import type {NumberFormat} from 'libphonenumber-js/core';

export interface MaskitoPhoneParams {
    // [...]
    format?: Extract<NumberFormat, 'INTERNATIONAL' | 'NATIONAL'>;
}
```

(for self-descriptive code purpose)

File: projects/demo/src/pages/phone/phone-doc.template.html [hunk 3]
```diff
@@ -130,6 +130,40 @@
...
+                    .
+                </p>
+                <p>
+                    <strong>Note:</strong>
+                    National format only works with
+                    <code>strict: true</code>
+                    mode (requires
+                    <code>countryIsoCode</code>
+                    ).
+                </p>
```
Comment (nsbarsukov) [inline projects/demo/src/pages/phone/phone-doc.template.html]:
```suggestion
                <tui-notification size="m">
                    National format only works with defined
                    <code>countryIsoCode</code>
                    property!
                </tui-notification>
```

File: projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts [hunk 4]
```diff
@@ -9,11 +15,19 @@ import {getCountryCallingCode, parsePhoneNumber} from 'libphonenumber-js/core';
...
+    format = 'INTERNATIONAL',
 }: {
     countryIsoCode: CountryCode;
     metadata: MetadataJson;
+    /**
+     * Phone number format.
+     * - 'INTERNATIONAL' (default): Includes country code prefix
+     * - 'NATIONAL': Country-specific format without country code
+     */
+    format?: PhoneNumberFormat;
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts]: No need to repeat this code comment always.
The single description inside `MaskitoPhoneParams` is enough

File: projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts [hunk 5]
```diff
@@ -9,11 +15,19 @@ import {getCountryCallingCode, parsePhoneNumber} from 'libphonenumber-js/core';
...
 }): MaskitoPreprocessor {
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts]: Use the such pattern (is is already used inside source code of `Number` mask) for all postprocessor of `Phone` mask

```ts
export function cutInitCountryCodePreprocessor({
    countryIsoCode,
    metadata,
    format,
}: Required<
    Pick<MaskitoPhoneParams, 'countryIsoCode' | 'format' | 'metadata'>
>): MaskitoPreprocessor {
```

File: projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts [hunk 6]
```diff
@@ -9,11 +15,19 @@ import {getCountryCallingCode, parsePhoneNumber} from 'libphonenumber-js/core';
...
     let isInitializationPhase = true;
+    const isNational = format === 'NATIONAL';
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts]: Is it used only once inside `if(isNational) {}` block below?
Just inline it then

File: projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts [hunk 7]
```diff
@@ -26,7 +40,46 @@ export function cutInitCountryCodePreprocessor({
...
+                try {
+                    const {nationalNumber} = parsePhoneNumber(
+                        value,
+                        countryIsoCode,
+                        metadata,
+                    );
+                    /**
+                     * Format the national number using country-specific formatting.
+                     */
+                    const formattedNational = formatIncompletePhoneNumber(
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts]: Is this constant is used only once ?
Just inline it then
```ts
{
    elementState: {
        selection,
        value: formatIncompletePhoneNumber(
            nationalNumber,
            countryIsoCode,
            metadata,
        ),
    },
}
```

File: projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/cut-init-country-code-preprocessor.ts]: We twice use `try`/`catch` + `parsePhoneNumber` to get the same result for `INTERNATIONAL` & `NATIONAL` formats.

Is it better to do early exit ?
```ts
/**
 * International format:
 * If the value already starts with the expected prefix (e.g., "+7 "),
 * don't reformat it. This avoids breaking selection positions when
 * the input already has a properly formatted value (e.g., an initial
 * value set on the element before Maskito attaches).
 *
 * National format:
 * If value starts with '+', extract national number.
 * Otherwise, assume it's already in national format.
 */
if (
    (format === 'INTERNATIONAL' && value.startsWith(code)) ||
    (format === 'NATIONAL' && !value.startsWith('+'))
) {
    return {elementState};
}
```

Then

```ts
try {
    const {nationalNumber} = parsePhoneNumber(value, countryIsoCode, metadata);

    return {
        elementState: {
            value: isNational
                ? formatIncompletePhoneNumber(
                      nationalNumber,
                      countryIsoCode,
                      metadata,
                  )
                : `${code} ${nationalNumber}`,
            selection,
        },
    };
} catch {
    return {elementState};
}
```

File: projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts [hunk 8]
```diff
@@ -2,35 +2,108 @@ import type {MaskitoPreprocessor} from '@maskito/core';
...
+
+/**
+ * Converts an international phone value to national format.
+ */
+function convertToNationalFormat(
+    value: string,
+    countryIsoCode: CountryCode,
+    metadata: MetadataJson,
+): string | null {
+    const formatter = new AsYouType({defaultCountry: countryIsoCode}, metadata);
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts]: Could it simplified to the following way ?


```suggestion
    const formatter = new AsYouType(countryIsoCode, metadata);
```

File: projects/phone/src/lib/masks/phone/utils/get-phone-template.ts [hunk 9]
```diff
@@ -1,16 +1,82 @@
...
+ * Generates a phone template string based on the current input value.
+ *
+ * @param formatter - AsYouType formatter instance
+ * @param value - Current input value
+ * @param separator - Separator character between number groups
+ * @param countryIsoCode - Country ISO code (required for national format)
+ * @param metadata - libphonenumber-js metadata
+ * @param format - Phone number format ('INTERNATIONAL' or 'NATIONAL')
+ * @returns Template string with 'x' placeholders for digits
+ */
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/utils/get-phone-template.ts]: Remove this js-doc. 
Comments inside `MaskitoPhoneParams` is enough

File: projects/phone/src/lib/masks/phone/utils/get-phone-template.ts [hunk 10]
```diff
@@ -1,16 +1,82 @@
...
+
+        return getNationalPhoneTemplate({
+            value: normalizedValue,
+            countryIsoCode,
+            metadata,
+            separator,
+        });
+    }
+
+    return getInternationalPhoneTemplate({formatter, value, separator});
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/utils/get-phone-template.ts:36]: Honestly, this file looks too overwhelmed. Are all new code lines really required ?

For example, why utility `getInternationalPhoneTemplate` contains additional logic if everything works even without them earlier ?  🤔

File: projects/demo-integrations/src/tests/component-testing/phone/phone-national-format.cy.ts [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+                expectedCursor: 8,
+            },
+            {
+                name: 'after area code',
+                leftArrows: 7,
+                expectedValue: '(212) 433-355',
+                expectedCursor: 6,
+            },
+        ],
+    },
```
Comment (nsbarsukov) [inline projects/demo-integrations/src/tests/component-testing/phone/phone-national-format.cy.ts]: I think we over-optimized for DRY here, which makes the test file hard to follow. Since tests also act as living documentation (not only bug-free ensurance), clarity is just as important as logic. Could you take a look at the other test files in the repo and refactor this one to match that level of readability?

File: projects/phone/src/lib/masks/phone/utils/get-phone-template.ts (hunk 10 above)
Comment (jbenning-ic) [reply projects/phone/src/lib/masks/phone/utils/get-phone-template.ts:36]: Simplified it - reduced from 155 → 103 lines by removing complex area code detection logic and verbose comments.

Why remaining code is needed:
- `getNationalPhoneTemplate` - `AsYouType.getTemplate()` only works for international. National format requires `formatIncompletePhoneNumber` + converting digits to template placeholders.
- Separator logic - Countries use different native separators (US/RU have dashes, FR has spaces). Need ~10 lines to handle both.
`hasDigitsOrPlus` guard - Edge case: '( ) -' → '', but '+' → 'x'. Tests cover this.

**Comment by jbenning-ic** [comment]
@nsbarsukov I hope you enjoyed your holidays! Thank you for reviewing this PR, I have addressed your comments, PR is ready for re-review when you find the time. Thanks, as always 🙏

File: projects/demo/src/pages/phone/phone-doc.template.html [hunk 12]
```diff
@@ -212,6 +243,35 @@
                     <code>-</code>
                 </p>
             </ng-template>
+            <ng-template
+                documentationPropertyMode="input"
+                documentationPropertyName="format"
+                documentationPropertyType="MaskitoPhoneParams['format']"
```
Comment (nsbarsukov) [inline projects/demo/src/pages/phone/phone-doc.template.html]: Let's inline it here (only for this part of documentation!)

```suggestion
                documentationPropertyType="'NATIONAL' | 'INTERNATIONAL'"
```
To be similar to other propertys of API pages (e.g. `metadata` property has `MetadataJson` type instead of `MaskitoPhoneParams['metadata']` – it is okay and descriptive for users)

File: projects/kit/src/lib/masks/date/utils/tests/stringify-date.spec.ts [hunk 13]
```diff
@@ -10,33 +10,33 @@ describe('maskitoStringifyDate', () => {
         [
             'dd/mm',
             [
-                {date: new Date('2004-02-01'), text: '01.02'},
-                {date: new Date('2012-12-30'), separator: '-', text: '30-12'},
+                {date: new Date('2004-02-01T12:00:00'), text: '01.02'},
+                {date: new Date('2012-12-30T12:00:00'), separator: '-', text: '30-12'},
```
Comment (nsbarsukov) [inline projects/kit/src/lib/masks/date/utils/tests/stringify-date.spec.ts]: It does not passes locally for your device ?

I am not against these change but lets do it in separate PR and provide PR body with error description

File: projects/phone/src/lib/masks/phone/phone-mask.ts [hunk 14]
```diff
@@ -4,19 +4,34 @@ import type {CountryCode, MetadataJson} from 'libphonenumber-js/core';
...
+    metadata: MetadataJson;
+    strict?: boolean;
+    separator?: string;
+    /**
+     * Phone number format.
+     * - 'INTERNATIONAL' (default): Includes country code prefix (e.g., +1 212 343-3355)
+     * - 'NATIONAL': Country-specific format without country code (e.g., (212) 343-3355).
+     *   Only works with strict mode (requires countryIsoCode).
+     */
+    format?: 'INTERNATIONAL' | 'NATIONAL';
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/phone-mask.ts]: https://github.com/taiga-family/maskito/pull/2461#discussion_r2690489940

File: projects/phone/src/lib/masks/phone/phone-mask-strict.ts [hunk 15]
```diff
@@ -15,22 +15,34 @@ export function maskitoPhoneStrictOptionsGenerator({
...
+    >): Required<MaskitoOptions> {
+    const isNational = format === 'NATIONAL';
     const code = getCountryCallingCode(countryIsoCode, metadata);
     const formatter = new AsYouType(countryIsoCode, metadata);
-    const prefix = `+${code} `;
+    /**
+     * For international format: prefix is "+{countryCode} " (e.g., "+1 ")
+     * For national format: prefix is empty string (no country code displayed)
+     */
+    const prefix = isNational ? '' : `+${code} `;
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/phone-mask-strict.ts]: excessive comment (it is already obvious from code line below)

File: projects/phone/src/lib/masks/phone/phone-mask-strict.ts [hunk 16]
```diff
@@ -15,22 +15,34 @@ export function maskitoPhoneStrictOptionsGenerator({
...
-}: {
-    countryIsoCode: CountryCode;
-    metadata: MetadataJson;
-    separator?: string;
-}): Required<MaskitoOptions> {
+    format = 'INTERNATIONAL',
+}: Pick<MaskitoPhoneParams, 'format' | 'separator'> &
+    Required<
+        Pick<MaskitoPhoneParams, 'countryIsoCode' | 'metadata'>
+    >): Required<MaskitoOptions> {
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/phone-mask-strict.ts]:
```suggestion
}: Omit<Required<MaskitoPhoneParams>, 'strict'>): Required<MaskitoOptions> {
```

File: projects/phone/src/lib/masks/phone/phone-mask-strict.ts [hunk 17]
```diff
@@ -43,19 +55,28 @@ export function maskitoPhoneStrictOptionsGenerator({
...
-            maskitoCaretGuard((value, [from, to]) => [
-                from === to ? prefix.length : 0,
-                value.length,
-            ]),
-        ],
+        plugins: isNational
+            ? []
+            : [
+                  maskitoCaretGuard((value, [from, to]) => [
+                      from === to ? prefix.length : 0,
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/phone-mask-strict.ts]: `prefix.length` always equals for 0 for national formatter.

I think we can revert this lines and we will get the same result (caret guard will be disabled).

File: projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts [hunk 18]
```diff
@@ -2,35 +2,97 @@ import type {MaskitoPreprocessor} from '@maskito/core';
...
+import type {MaskitoPhoneParams} from '../phone-mask';
+
+/**
+ * Converts an international phone value to national format.
+ */
+function convertToNationalFormat(
+    value: string,
+    countryIsoCode: CountryCode,
+    metadata: MetadataJson,
+): string | null {
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts]: Do we really need to add `null` to return type ?

Could we just use empty string for "error"-cases ?

File: projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts [hunk 19]
```diff
@@ -41,13 +103,29 @@ export function validatePhonePreprocessorGenerator({
             );
 
             if (!validationError || validationError === 'TOO_SHORT') {
-                // handle paste-event with different code, for example for 8 / +7
+                /**
+                 * Handle paste-event with different code, for example for 8 / +7.
+                 */
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts]: Why ?

File: projects/demo-integrations/src/tests/component-testing/phone/phone-national-format.cy.ts [hunk 20]
```diff
@@ -0,0 +1,373 @@
...
+            });
+
+            it('(212) 343-3355| => Backspace => (212) 343-335|', () => {
+                cy.get('input')
+                    .should('have.value', '(212) 343-3355')
+                    .focus()
+                    .type('{moveToEnd}')
+                    .type('{backspace}')
+                    .should('have.value', '(212) 343-335')
+                    .should('have.prop', 'selectionStart', 13)
```
Comment (nsbarsukov) [inline projects/demo-integrations/src/tests/component-testing/phone/phone-national-format.cy.ts]: Let's use this pattern to readability (for this and other cases of this file)


```suggestion
                    .should('have.prop', 'selectionStart', '(212) 343-335'.length)
```

File: projects/demo-integrations/src/tests/component-testing/phone/phone-national-format.cy.ts [hunk 21]
```diff
@@ -0,0 +1,373 @@
...
+                    .should('have.prop', 'selectionEnd', 13);
+            });
+
+            it('(212) 343|-3355 => Backspace => (212) 34|3-355', () => {
+                cy.get('input')
+                    .should('have.value', '(212) 343-3355')
+                    .focus()
+                    .type('{moveToEnd}')
+                    .type('{leftArrow}'.repeat(5))
```
Comment (nsbarsukov) [inline projects/demo-integrations/src/tests/component-testing/phone/phone-national-format.cy.ts]: The same for `.repeat(...)`
```suggestion
                    .type('{leftArrow}'.repeat('-3355'.length))
```

File: projects/kit/src/lib/masks/date/utils/tests/stringify-date.spec.ts (hunk 13 above)
Comment (nsbarsukov) [reply projects/kit/src/lib/masks/date/utils/tests/stringify-date.spec.ts]: And revert this file for this PR

**Comment by nsbarsukov** [comment]
Looks like a bug

https://github.com/user-attachments/assets/715969d9-976b-4215-8a20-825dfdd00b88

**Comment by jbenning-ic** [comment]
> Looks like a bug
> 
>  bug.mov

Should be fixed now! Added unit tests to cover this as well

File: projects/demo/src/pages/phone/examples/6-national-format/component.ts [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+    protected value = '';
+    protected readonly mask = mask;
+
+    /**
+     * Pattern for iOS Safari to allow phone number input.
+     * National format doesn't include '+', so pattern is different.
+     */
+    protected get pattern(): string {
+        return this.isApple ? '[0-9()-]{1,20}' : '';
+    }
```
Comment (nsbarsukov) [inline projects/demo/src/pages/phone/examples/6-national-format/component.ts]: `this.isApple` is always the same because it DI-token value (no run-time changes).

Could it be simplified to just static property ?
```ts
protected readonly pattern = this.isApple ? '[0-9()-]{1,20}' : '';
```

File: projects/demo/src/pages/phone/examples/6-national-format/mask.ts [hunk 23]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * National format phone mask for United States.
+ * Displays phone numbers in national format: (XXX) XXX-XXXX
+ * without the country code prefix.
+ */
+export default maskitoPhoneOptionsGenerator({
+    countryIsoCode: 'US',
+    metadata,
+    format: 'NATIONAL',
```
Comment (nsbarsukov) [inline projects/demo/src/pages/phone/examples/6-national-format/mask.ts:12]: We can still enter number with country prefix.

Is it acceptable for enabled `NATIONAL` format? 🤔 

Image

File: projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts [hunk 24]
```diff
@@ -2,35 +2,97 @@ import type {MaskitoPreprocessor} from '@maskito/core';
...
+                 * International format autocomplete handling.
+                 */
+                const formatter = new AsYouType(countryIsoCode, metadata);
 
-            formatter.reset();
+                formatter.input(value);
+                const numberValue = formatter.getNumberValue() ?? '';
 
-            return {elementState: {value: formatter.input(numberValue), selection}};
+                formatter.reset();
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/processors/validate-phone-preprocessor.ts]: These lines repeat twice:
* Here
* And inside `convertToNationalFormat` utility above

Should we create local utility inside this file ? 🤔

File: projects/phone/src/lib/masks/phone/tests/phone-mask.spec.ts [hunk 25]
```diff
@@ -60,4 +60,88 @@ describe('Phone (maskitoTransform)', () => {
...
+            });
+
+            /**
+             * Note: When input starts with '8', libphonenumber-js interprets it
+             * as part of the number rather than the old Russian trunk prefix.
+             * We test that the national number portion is extracted correctly.
+             */
+            it('strips alternate country code (8) via parsing', () => {
+                expect(maskitoTransform('89202800155', options)).toBe(
+                    '8 (920) 280-01-55',
```
Comment (nsbarsukov) [inline projects/phone/src/lib/masks/phone/tests/phone-mask.spec.ts:125]: @taiga-family/core-team Is it acceptable behavior for us ?

File: projects/demo/src/pages/phone/examples/6-national-format/mask.ts (hunk 23 above)
Comment (nsbarsukov) [reply projects/demo/src/pages/phone/examples/6-national-format/mask.ts:12]: @taiga-family/core-team probably no ?

**Comment by nsbarsukov** [comment]
@jbenning-ic Thanks for promptly addressing the review comments. The new codebase looks promising and has a good chance of getting merged. We just need to discuss the final details.

File: projects/demo/src/pages/phone/examples/6-national-format/mask.ts (hunk 23 above)
Comment (jbenning-ic) [reply projects/demo/src/pages/phone/examples/6-national-format/mask.ts:12]: [libphonenumber-js](https://catamphetamine.gitlab.io/libphonenumber-js/?parseCountry=US&asYouTypeCountry=US&asYouTypeValue=12125551234&findNumbersCountry=US&findNumbersText=For%20more%20details%20call%20%2B7%20(800)%20555-35-35%20internationally%20or%20reach%20the%20local%20US%20branch%20at%20(213)%20373-4253%20ext.%201234%20for%20paid%20customer%20support.) allows it

image
image

File: projects/phone/src/lib/masks/phone/tests/phone-mask.spec.ts (hunk 25 above)
Comment (nsbarsukov) [reply projects/phone/src/lib/masks/phone/tests/phone-mask.spec.ts:125]: We discussed this case. It is too difficult to detect all similar national trunk prefixes. 
Even if we managed to detect them, manually transforming the value is too fragile. Let's leave it as is.
